### PR TITLE
ArangoDB version in membership status

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: pkg/deployment/resources/pod_creator.go [hunk 1]
```diff
@@ -662,11 +662,21 @@ func (r *Resources) createPodForMember(spec api.DeploymentSpec, memberID string,
...
 		finalizers := r.createPodFinalizers(group)
-		if err := k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
+		var p v1.Pod
+		if p, err = k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
 			engine, requireUUID, terminationGracePeriod, args, env, finalizers, livenessProbe, readinessProbe, tolerations, serviceAccountName, tlsKeyfileSecretName, rocksdbEncryptionSecretName,
 			clusterJWTSecretName, groupSpec.GetNodeSelector(), groupSpec.PriorityClassName, groupSpec.Resources, exporter, groupSpec.GetSidecars(), groupSpec.VolumeClaimTemplate); err != nil {
 			return maskAny(err)
 		}
+
+		if err == nil {
```
Comment (joerg84) [inline pkg/deployment/resources/pod_creator.go]: Do we need this condition? (no objection)

Comment (kvahed) [reply pkg/deployment/resources/pod_creator.go]: yes

Comment (joerg84) [reply pkg/deployment/resources/pod_creator.go]: the line above is ```err != nil {
			return maskAny(err)
		}```

Comment (kvahed) [reply pkg/deployment/resources/pod_creator.go]: you win

File: pkg/deployment/resources/pod_creator.go [hunk 2]
```diff
@@ -662,11 +662,15 @@ func (r *Resources) createPodForMember(spec api.DeploymentSpec, memberID string,
 		engine := spec.GetStorageEngine().AsArangoArgument()
 		requireUUID := group == api.ServerGroupDBServers && m.IsInitialized
 		finalizers := r.createPodFinalizers(group)
-		if err := k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
+		if err = k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
```
Comment (joerg84) [inline pkg/deployment/resources/pod_creator.go]: Do we still need this changed line here?

File: pkg/deployment/resources/pod_creator.go [hunk 3]
```diff
@@ -662,11 +662,15 @@ func (r *Resources) createPodForMember(spec api.DeploymentSpec, memberID string,
...
 			engine, requireUUID, terminationGracePeriod, args, env, finalizers, livenessProbe, readinessProbe, tolerations, serviceAccountName, tlsKeyfileSecretName, rocksdbEncryptionSecretName,
 			clusterJWTSecretName, groupSpec.GetNodeSelector(), groupSpec.PriorityClassName, groupSpec.Resources, exporter, groupSpec.GetSidecars(), groupSpec.VolumeClaimTemplate); err != nil {
 			return maskAny(err)
 		}
+
+		m.ArangoVersion = status.CurrentImage.ArangoDBVersion
+		status.Members.Update(m, group)
```
Comment (maierlars) [inline pkg/deployment/resources/pod_creator.go]: This Update is not necessary.

File: pkg/deployment/resources/pod_creator.go [hunk 4]
```diff
@@ -662,11 +662,15 @@ func (r *Resources) createPodForMember(spec api.DeploymentSpec, memberID string,
...
 		requireUUID := group == api.ServerGroupDBServers && m.IsInitialized
 		finalizers := r.createPodFinalizers(group)
-		if err := k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
+		if err = k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
 			engine, requireUUID, terminationGracePeriod, args, env, finalizers, livenessProbe, readinessProbe, tolerations, serviceAccountName, tlsKeyfileSecretName, rocksdbEncryptionSecretName,
 			clusterJWTSecretName, groupSpec.GetNodeSelector(), groupSpec.PriorityClassName, groupSpec.Resources, exporter, groupSpec.GetSidecars(), groupSpec.VolumeClaimTemplate); err != nil {
 			return maskAny(err)
 		}
+
+		m.ArangoVersion = status.CurrentImage.ArangoDBVersion
```
Comment (maierlars) [inline pkg/deployment/resources/pod_creator.go]: Maybe we should store the _image_, not the version. You can always convert Image->Version.

Comment (maierlars) [reply pkg/deployment/resources/pod_creator.go]: Maybe this should be generic code. You could move this up to the place where the image info is selected. Doing so would also set the image for Syncer Pods.

File: pkg/deployment/resources/pod_creator.go (hunk 2 above)
Comment (kvahed) [reply pkg/deployment/resources/pod_creator.go]: Nope.done

File: pkg/deployment/resources/pod_creator.go (hunk 4 above)
Comment (kvahed) [reply pkg/deployment/resources/pod_creator.go]: - one too many update ermoved
- the image id added
- the code is generic, it is correct version and image id of newly created pod

**Comment by joerg84** [comment]
> LGTM. We go without a test, since this is a very simple change.

Tests are still useful to prevent regressions and enable refactoring

**Pull request merged**